### PR TITLE
feat(var-naming): restore golint behavior by default, backwards-compatible rename and invert skipPackageNameCollisionWithGoStd

### LINKED ISSUE #1602: var-naming: Go stdlib collision check breaks defaulting to golint behavior
**Describe the bug**
The `RULES_DESCRIPTIONS.md` promises about the `var-naming` rule:

> By default, the rule behaves exactly as the alternative in golint but optionally, you can relax it (see https://github.com/golang/lint/issues/89).

However, the new v1.13.0 behavior of `skipPackageNameCollisionWithGoStd` conflicts with that promise, making our behavior stricter than the original.

Given the following folder structure:
```text
http/
    http.go
```

And given the following content of `http.go`:
```go
# Package http contains helpers.
package http
```

The original behavior of `golint` gives no errors.

```bash
$ go install golang.org/x/lint/golint@latest
$ cd http
$ golint .
```

**To Reproduce**
1. I updated revive `go install github.com/mgechev/revive@latest`
2. I run it with no flags & no configuration file, inside the `http` folder described above:

```shell
revive .
```

**Expected behavior**
No errors.

**Logs**

> http.go:2:9: avoid package names that conflict with Go standard library package names

**Desktop (please complete the following information):**
 - OS: Fedora 43
 - Version of Go: 1.25.4

**Additional context**
The last version of revive with the expected behavior is v1.12.0.

### PR DESCRIPTION
**Pull request by chaimleib**
Fixes #1602.

# Motivation

As described in #1602, an empty config should result in behavior like golint, according to the [RULES_DESCRIPTIONS.md](https://github.com/mgechev/revive/blob/e174d6d752cbcd4db5d2a088deed60f40f34aa38/RULES_DESCRIPTIONS.md?plain=1#L1774) for `var-naming`:

> By default, the rule behaves exactly as the alternative in `golint` but optionally, you can relax it (see [golint/lint/issues/89](https://github.com/golang/lint/issues/89)).

This contract broke between v1.12.0 and v1.13.0 when `skipPackageNameCollisionWithGoStd` was added in #1540.

# Changes

* Replace {skip->check}PackageNameCollisionWithGoStd
  * Invert the logic of that flag accordingly.
  * Update RULES_DESCRIPTIONS.md to reflect the change.
  * Fix existing tests.
  * Add a new test that an empty config behaves like `golint`, skipping the check on package name collisions with the Go stdlibs.
* feat: Deprecate and gracefully translate `skipPackageNameCollisionWithGoStd`, if provided, with a warning.
  * To show the deprecation message, allow logging at `slog.LevelWarn` and above, even when not in DEBUG mode.
  * When in DEBUG mode, `slog.LevelDebug` messages were being suppressed, since `slog.LevelDebug < 0`, with 0 being the default and equal to `slog.LevelInfo`. Now DEBUG mode will emit debug-level messages.

# Testing

## Makefile targets

`make lint test` passes.

## Comparison with golint

Given the following folder structure:
```text
http/
    http.go
```

And given the following content of `http.go`:
```go
# Package http contains helpers.
package http
```

The original behavior of `golint` gives no errors.

```bash
$ go install golang.org/x/lint/golint@latest
$ cd http
$ golint .
```

The current `master` branch at `e174d6d` gives an error:

```bash
$ cd path/to/revive
$ git switch master
$ make build
$ cd ../http
$ ../revive/revive .
http.go:2:9: avoid package names that conflict with Go standard library package names
```

The modified `revive` gives no errors.

```bash
$ cd ../revive
$ git switch fix/var-naming/1602-default-golint-behavior
$ make build
$ cd ../http
$ ../revive/revive .
```

## Enabling the check

Build the modified `revive`.

```bash
$ cd ../revive
$ git switch fix/var-naming/1602-default-golint-behavior
$ make build
```

Make the http dir like so:
```text
http/
    http.go
    revive.toml
```

`http.go`
```go
# Package http contains helpers.
package http
```

`revive.toml`
```toml
enableAllRules = false

[rule.var-naming]
  arguments = [ [], [], [{ checkPackageNameCollisionWithGoStd = true }] ]
```

An error should appear when run.
```bash
$ cd ../http
$ ../revive/revive .
http.go:2:9: avoid package names that conflict with Go standard library package names
```

## Logging tests

All the below tests start from this basic state. Create a folder named `http` like so:

```text
http/
    http.go
    revive.toml
```

`http.go`

```go
# Package http contains helpers.
package http
```

`revive.toml`

```toml
enableAllRules = false

[rule.var-naming]
  arguments = []
```

Build `revive`:

```bash
$ cd path/to/revive
$ make build
```

Navigate to the `http` folder:

```bash
$ cd path/to/http
```

## Empty arguments

```bash
$ path/to/revive/revive .
$ ls
http.go  revive.toml
```

No errors, just like golint.

## Debug mode, empty arguments

```bash
$ DEBUG=y path/to/revive/revive .
time=2025-12-05T01:46:23.699-07:00 level=INFO msg="Logger initialized" logFile=revive.log
time=2025-12-05T01:46:23.699-07:00 level=INFO msg="Config loaded" rules=[var-naming]

$ ls
http.go  revive.log  revive.toml

$ cat revive.log
time=2025-12-05T01:46:23.699-07:00 level=INFO msg="Logger initialized" logFile=revive.log
time=2025-12-05T01:46:23.699-07:00 level=INFO msg="Config loaded" rules=[var-naming]
```

The command creates `revive.log`. No errors, just INFOs.

## With deprecated skip option true

Edit `revive.toml`:

```toml
enableAllRules = false

[rule.var-naming]
  arguments = [ [], [], [{ skipPackageNameCollisionGoStd = true }]]
```

```bash
$ path/to/revive/revive .
time=2025-12-05T01:40:48.454-07:00 level=WARN msg="an option in the configuration is deprecated, attempting to interpret it and continue" deprecated_option=skipPackageNameCollisionWithGoStd suggestion="instead, use its inverse, checkPackageNameCollisionWithGoStd" interpreted_as=map[checkPackageNameCollisionWithGoStd:false]
```

Expect no errors, just a deprecation warning.

## With deprecated skip option false

Edit `revive.toml`:

```toml
enableAllRules = false

[rule.var-naming]
  arguments = [ [], [], [{ skipPackageNameCollisionGoStd = false }]]
```

```bash
$ ../revive/revive .
time=2025-12-05T01:42:32.252-07:00 level=WARN msg="an option in the configuration is deprecated, attempting to interpret it and continue" deprecated_option=skipPackageNameCollisionWithGoStd suggestion="instead, use its inverse, checkPackageNameCollisionWithGoStd" interpreted_as=map[checkPackageNameCollisionWithGoStd:true]
http.go:2:9: avoid package names that conflict with Go standard library package names
```

Expect a deprecation warning and a lint error.

## With new check option false

Edit `revive.toml`:

```toml
enableAllRules = false

[rule.var-naming]
  arguments = [ [], [], [{ checkPackageNameCollisionGoStd = false }]]
```

```bash
$ path/to/revive/revive .
```

Expect no errors.

### CODE REVIEW

**Comment by chaimleib** [comment]
I marked these commits as "feat" instead of "fix" because the config flags have changed. In semver, I think this merits incrementing the minor version in major.minor.patch, rather than the patch version.

**Comment by chaimleib** [comment]
Oops, the deprecation warning wasn't printing. I reverted that part, and I'm working on that in a separate branch based off of this.

**Comment by chaimleib** [comment]
Now the deprecation works. Users will not see breakage with their configs if this gets merged, just a deprecation warning.

File: rule/var_naming.go [hunk 1]
```diff
@@ -117,11 +118,25 @@ func (r *VarNamingRule) Configure(arguments lint.Arguments) error {
 					r.extraBadPackageNames[strings.ToLower(n)] = struct{}{}
 				}
 			case isRuleOption(k, "skipPackageNameCollisionWithGoStd"):
-				r.skipPackageNameCollisionWithGoStd = fmt.Sprint(v) == "true"
+				r.checkPackageNameCollisionWithGoStd = fmt.Sprint(v) != "true"
+
+				slogger, err := logging.GetLogger()
+				if err == nil {
+					slogger.Warn(
+						"an option in the configuration is deprecated, attempting to interpret it and continue",
```
Comment (chavacava) [inline rule/var_naming.go]:
```suggestion
						"Deprecated configuration option for rule var-naming",
```
